snekbox/memfs.py

```diff
@@ -0,0 +1,151 @@
+"""Memory filesystem for snekbox."""
+from __future__ import annotations
+
+import logging
+import subprocess
+from collections.abc import Generator
+from contextlib import contextmanager
+from pathlib import Path
+from shutil import rmtree
+from threading import BoundedSemaphore
+from types import TracebackType
+from typing import Type
+from uuid import uuid4
+
+from snekbox.snekio import FileAttachment
+
+log = logging.getLogger(__name__)
+
+NAMESPACE_DIR = Path("/memfs")
+NAMESPACE_DIR.mkdir(exist_ok=True)
+NAMESPACE_DIR.chmod(0o711)  # Execute only access for other users
+
+
+def mount_tmpfs(name: str, size: int | str) -> Path:
+    """Create and mount a tmpfs directory."""
+    tmp = NAMESPACE_DIR / name
+    tmp.mkdir()
+    tmp.chmod(0o711)
+    # Mount the tmpfs
+    subprocess.check_call(
+        [
+            "mount",
+            "-t",
+            "tmpfs",
+            "-o",
+            f"size={size}",
+            "tmpfs",
+            str(tmp),
+        ]
+    )
+    return tmp
+
+
+def unmount_tmpfs(name: str) -> None:
+    """Unmount and remove a tmpfs directory."""
+    tmp = NAMESPACE_DIR / name
+    subprocess.check_call(["umount", str(tmp)])
+    rmtree(tmp, ignore_errors=True)
```

Comment:
It seems from my testing it still leaves the folder, should I do a `tmp.unlink()` instead then?

<img width="350" alt="image" src="https://user-images.githubusercontent.com/13956642/202793747-edcfc299-d642-42c3-a187-090aaf6f50f6.png">

Reply:
Ah, the directory left isn't the tmpfs, it's just the mountpoint directory created in `mount_tmpfs`. I think it would be better to use `tmp.rmdir()` and add a comment to make that clear.

Reply:
Resolved in [09bbc4f](https://github.com/python-discord/snekbox/pull/159/commits/09bbc4f11f32d91dfa20aeb3da77a5e2a054f49d)